python/tvm/relay/op/contrib/dnnl.py

```diff
@@ -0,0 +1,72 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=invalid-name, unused-argument
+"""DNNL library supported operators.
+There are two ways to registering a function for an op to indicate if it is
+supported by DNNL.
+
+- The first and simplest way is to use the helper so that
+users only need to provide the operator name and a boolean value to indicate if
+it is supported. For example:
+
+    .. code-block:: python
+
+      add = _register_external_op_helper("add")
+      add = _register_external_op_helper("add", True)
+      add = _register_external_op_helper("add", False)
+
+- The other way is to implement the function by themselves to
+check the attributes of the op and decide if it should be offloaded to DNNL.
+"""
+from ... import op as reg
+
+
+def _register_external_op_helper(op_name, supported=True):
```

Comment:
You can only use `reg.register` if you want. This is just an optional use case to reduce the lines of code. We will update the tutorial to explain this part.

Reply:
I see that it's optional, but is there a way we can make _register_external_op_helper a generic helper, not dnnl specific. I imagine other codegens will want to use the same thing and each one will essentially rewrite this function.

Reply:
I think this is consistent to what topi currently does. @comaniac can speak more about it

Reply:
Yes this is consistent to TOPI. Also, since this is the only utility/helper function for codegen annotation at this moment, we do not want to introduce another level of abstractions to developers (one more function call stack or a base class).  This is also based on the RFC discussion. We will mention this convenient approach in the tutorial and let developers choose. I think it's not a big deal for each codegen to rewrite this small helper function.

Reply:
OK, sounds reasonable.